### PR TITLE
Fix/location list activity test

### PR DESCRIPTION
**Pull request by llvasconcellos**

### CODE REVIEW

File: app/src/androidTest/java/org/projectbuendia/client/ui/lists/LocationListActivityTest.java [hunk 1]
```diff
@@ -55,31 +55,33 @@ public void testZonesAndTentsDisplayed_afterSettingsView() {
...
-        inUserLoginInitDemoPatient();
-        inUserLoginGoToLocationSelection();
-        inLocationSelectionClickLocation(LOCATION_NAME);
-        inPatientListClickFirstPatient(); // open patient chart
-
-        pressBack(); // back to search fragment
-        pressBack(); // back to tent selection screen
+        inUserLoginGoToPatientCreation();
+        screenshot("Test Start");
+        String id = generateId();
```
Comment (G1DR4) [inline app/src/androidTest/java/org/projectbuendia/client/ui/lists/LocationListActivityTest.java:60]: you don`t need id in any other place, generate it inside populateNewPatientFields()

File: app/src/androidTest/java/org/projectbuendia/client/ui/FunctionalTestCase.java [hunk 2]
```diff
@@ -437,4 +441,37 @@ protected void inPatientListClickPatientWithId(String id) {
             .inAdapterView(hasId(R.id.fragment_patient_list))
             .atPosition(0));
     }
+
+    /** Generates IDs to identify the newly created patient. */
+    protected String generateId() {
+        String id = Long.toString(new Date().getTime()%100000);
+        return id;
```
Comment (zestyping) [inline app/src/androidTest/java/org/projectbuendia/client/ui/FunctionalTestCase.java]: id is only used once; simplify to just `return <expr>`

`"" + x` is more concise than `Long.toString(x)`

**Comment by zestyping** [comment]
The changes to make LocationListActivityTest pass look good. After separating these changes from the patient validation code (which shouldn't go in yet), feel free to merge.

**Comment by zestyping** [comment]
Note, the reason not to commit the patient validation code is to save time.  If it will actually take less time to commit these as is without removing the patient validation code, it's okay to proceed with the more expedient option, with the understanding that the patient validation code will be backed out later.

File: app/src/androidTest/java/org/projectbuendia/client/ui/lists/LocationListActivityTest.java (hunk 1 above)
Comment (llvasconcellos) [inline app/src/androidTest/java/org/projectbuendia/client/ui/lists/LocationListActivityTest.java:60]: Actually I need it in other tests.

File: app/src/androidTest/java/org/projectbuendia/client/ui/FunctionalTestCase.java (hunk 2 above)
Comment (llvasconcellos) [inline app/src/androidTest/java/org/projectbuendia/client/ui/FunctionalTestCase.java]: Done.

**Pull request merged**